doc/plugins_readme.md

```diff
@@ -0,0 +1,135 @@
+
+```cpp
+// Example of a plugin with description "Mock library".
+// The plugin exports two classes - MockClass and AnotherMockClass.
+// MockBaseClass is the parent class for the two exported classes.
+// "mock_class" and "another_mock_class" are identifiers which are used as keys
+// for the std::unordered_map used by this Manifest object under the hood.
+BEGIN_MANIFEST("Mock library")
+EXPORT(MockClass, MockBaseClass, "mock_class")
+EXPORT(AnotherMockClass, MockBaseClass, "another_mock_class")
+END_MANIFEST()
+
+// The above macros are translated to:
+
+// BEGIN_MANIFEST("Mock library")
+extern "C" Manifest shogunManifest() {
+    static Manifest manifest("Mock library", {
+        // EXPORT(MockClass, MockBaseClass, "mock_class")
+        std::make_pair("mock_class", erase_type(
+            MetaClass<MockBaseClass>(erase_type(
+                std::function<Some<MockBaseClass>()>(
+                    []() -> Some<MockBaseClass>
+                    {
+                        return Some<MockBaseClass>(new MockClass);
+                    }
+                ))))),
+        // EXPORT(AnotherMockClass, MockBaseClass, "another_mock_class")
+        std::make_pair("another_mock_class", erase_type(
+            MetaClass<MockBaseClass>(erase_type(
+                std::function<Some<MockBaseClass>()>(
+                    []() -> Some<MockBaseClass>
+                    {
+                        return Some<MockBaseClass>(new AnotherMockClass);
+                    }
+                ))))),
+    // END_MANIFEST()
+    });
+    return manifest;
+}
+```
+#### Example
+
+This example illustrates how to load and use a Shogun plugin.
+
+```cpp
+// Load a plugin named "libplugin.so" as a Library class instance.
+Library library = load_library("./libplugin.so");
+
+// Get the manifest, every plugin should have a manifest compatible with Shogun's requirements.
+Manifest manifest = library.manifest();
+
+// Get MetaClass objects of MockClass and AnotherMockClass, both of which are derived from MockBaseClass.
+MetaClass<MockBaseClass> mock_class = manifest.class_by_name<MockBaseClass>("mock_class");
+MetaClass<MockBaseClass> another_mock_class = manifest.class_by_name<MockBaseClass>("another_mock_class");
+
+// MetaClass objects return shared-pointer-like objects (Some<MockBaseClass>) for classes available in libplugin.so.
+Some<MockBaseClass> mock_class_obj = mock_class.instance();
+Some<MockBaseClass> another_mock_class_obj = another_mock_class.instance();
+
+// Now the objects can be used as desired.
+std::string mock_class_name = mock_class_obj->get_name();
+std::string another_mock_class_name = another_mock_class_obj->get_name();
+mock_class_obj->mock_method();
+another_mock_class_obj->mock_method;
+```
+
+### Tests
+
+Unit-tests for this plugin architecture can found in `tests/unit/base/Plugin_unittest.cc`. The unit-tests use `tests/unit/base/MockBaseClass.h` and a dummy plugin `tests/unit/base/MockLibrary.cpp`.
```

Comment:
I haven't linked these files since these are not yet merged in develop but live on feature/tags branch. But I'm confused, shall I link them now or later (after this gets merged in develop)?